client/my-sites/site-settings/form-seo.jsx

```diff
@@ -0,0 +1,257 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import Button from 'components/button';
+import SectionHeader from 'components/section-header';
+import ExternalLink from 'components/external-link';
+import Notice from 'components/notice';
+import NoticeAction from 'components/notice/notice-action';
+import protectForm from 'lib/mixins/protect-form';
+import FormInput from 'components/forms/form-text-input-with-affixes';
+import FormInputValidation from 'components/forms/form-input-validation';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormSettingExplanation from 'components/forms/form-setting-explanation';
+import get from 'lodash/get';
+import notices from 'notices';
+
+export default React.createClass( {
+	displayName: 'SiteSettingsFormSEO',
+
+	mixins: [ protectForm.mixin ],
+
+	getInitialState() {
+		const { site } = this.props;
+		return {
+			blog_public: 1,
+			seo_meta_description: get( site, 'options.seo_meta_description', '' ),
+			verification_services_codes: get( site, 'options.verification_services_codes', null )
+		};
+	},
+
+	handleMetaChange( event ) {
+		this.setState( { seo_meta_description: get( event, 'target.value', '' ) } );
+	},
+
+	handleVerificationCodeChange( event ) {
+		const productId = get( event, 'target.id', '' ).replace( 'verification_code_', '' );
+
+		// Build verification services object to send back to API
+		let verificationServicesCodes = this.getCurrentVerificationCodes();
+
+		if ( ! verificationServicesCodes.hasOwnProperty( productId ) ) {
+			return;
+		}
+
+		verificationServicesCodes[ productId ] = get( event, 'target.value', '' );
+
+		this.setState( { verification_services_codes: verificationServicesCodes } );
+	},
+
+	getCurrentVerificationCodes() {
+		return {
+			google: this.refs.verification_code_google.props.value,
```

Comment:
We shouldn't have to rely on refs inspection for these values, it'd be better to have these set in component state directly on onChange or in redux.